src/main/java/io/supertokens/webserver/api/dashboard/DashboardUserAPI.java

```diff
@@ -0,0 +1,207 @@
+                email = normalizeStringParam(email, "email");
+                Dashboard.updateUsersCredentialsWithEmail(main, email, newEmail, newPassword);
+
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "OK");
+                super.sendJsonResponse(200, response, resp);
+            }
+        } catch (DuplicateEmailException e) {
+            JsonObject response = new JsonObject();
+            response.addProperty("status", "EMAIL_ALREADY_EXISTS_ERROR");
+            super.sendJsonResponse(200, response, resp);
+
+        } catch (StorageQueryException | StorageTransactionLogicException e) {
+            throw new ServletException(e);
+        }
+
+        JsonObject response = new JsonObject();
+        response.addProperty("status", "OK");
+        super.sendJsonResponse(200, response, resp);
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject input = InputParser.parseJsonObjectOrThrowError(req);
+
+        String userId = InputParser.parseStringOrThrowError(input, "userId", true);
+        try {
+            if (userId != null) {
+                // normalize userId
+                userId = normalizeStringParam(userId, "userId");
+                boolean didUserExist = Dashboard.deleteUserWithUserId(main, userId);
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "OK");
+                response.addProperty("didUserExist", didUserExist);
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+
+            String email = InputParser.parseStringOrThrowError(input, "email", true);
+
+            if (email != null) {
+                // normalize email
+                email = normalizeStringParam(email, "email");
+                boolean didUserExist = Dashboard.deleteUserWithEmail(main, email);
+                JsonObject response = new JsonObject();
+                response.addProperty("status", "OK");
+                response.addProperty("didUserExist", didUserExist);
+                super.sendJsonResponse(200, response, resp);
+                return;
+            }
+
+        } catch (StorageQueryException e) {
+            throw new ServletException(e);
+        }
+
+        JsonObject response = new JsonObject();
+        response.addProperty("status", "OK");
+        response.addProperty("didUserExist", false);
+        super.sendJsonResponse(200, response, resp);
+    }
+
+    private static String normalizeStringParam(String param, String paramName) throws ServletException {
+        param = param.trim();
+        if (param.length() == 0) {
+            throw new ServletException(
+                    new WebserverAPI.BadRequestException("Field name " + paramName + " cannot be an empty String"));
+        }
+        return param;
+    }
```

Comment:
Shouldnt this be a util function for all classes in WebServer? Feels generic enough to be required by multiple API classes